engine/src/main/java/com/datatorrent/stram/engine/OperatorContext.java

```diff
@@ -84,12 +87,13 @@ public void setIdleTimeout(long idleTimeout)
    * @param attributes the value of attributes
    * @param parentContext
    */
-  public OperatorContext(int id, AttributeMap attributes, Context parentContext)
+  public OperatorContext(int id, String name, AttributeMap attributes, Context parentContext)
   {
     super(attributes, parentContext);
     this.lastProcessedWindowId = Stateless.WINDOW_ID;
     this.id = id;
     this.stateless = super.getValue(OperatorContext.STATELESS);
+    this.name = Preconditions.checkNotNull(name, "operator name");
```

Comment:
However, having this check here it ensures that later no classes instantiating operatorContext do not pass a null name.